CMakeLists.txt

```diff
@@ -4,186 +4,256 @@
 
 CMAKE_MINIMUM_REQUIRED(VERSION 3.5.2 FATAL_ERROR)
 
-### CMP0025   Compiler id for Apple Clang is now AppleClang.
-### CMP0042   MACOSX_RPATH is enabled by default.
+# Set cmake policy by version: https://cmake.org/cmake/help/latest/manual/cmake-policies.7.html
+if(${CMAKE_VERSION} VERSION_LESS 3.12)
+    cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
+else()
+    cmake_policy(VERSION 3.12)
+endif()
 
-FOREACH (p
-        CMP0025 # CMake 3.0
-        CMP0042 # CMake 3.0
-        )
-    IF (POLICY ${p})
-        cmake_policy(SET ${p} NEW)
-    ENDIF ()
-endforeach ()
+# Define project
+project(Monitoring
+  VERSION 1.7.2
+  DESCRIPTION "O2 Monitoring library"
+  LANGUAGES CXX
+)
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake) # project specific cmake dir
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
+# Documentation dir
+add_subdirectory(doc)
 
-project(Monitoring)
+# Add compiler flags for warnings and (more importantly) fPIC and debug symbols
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic -Wextra -fPIC")
```

Comment:
For `-fPIC` there is a CMake abstraction available [`CMAKE_POSITION_INDEPENDENT_CODE`](https://cmake.org/cmake/help/v3.4/variable/CMAKE_POSITION_INDEPENDENT_CODE.html).

Reply:
Indeed